qualtran/bloqs/basic_gates/z_basis.py

```diff
@@ -262,13 +283,67 @@ def as_cirq_op(
     def _t_complexity_(self) -> 'TComplexity':
         return TComplexity(clifford=1)
 
+    def wire_symbol(
+        self, reg: Optional['Register'], idx: Tuple[int, ...] = tuple()
+    ) -> 'WireSymbol':
+        if reg is None:
+            return Text('')
+
+        return TextBox('Z')
+
 
 @bloq_example
 def _zgate() -> ZGate:
     zgate = ZGate()
     return zgate
 
 
+@frozen
+class CZ(Bloq):
+    """Two-qubit controlled-Z gate.
+
+    Registers:
+        ctrl: One-bit control register.
+        target: One-bit target register.
+    """
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        return Signature.build(q1=1, q2=1)
```

Comment:
Should we just accept `Signature([Register('q', QBit(), shape=2)])` instead of a `q1` and `q2` ? This will keep things consistent with rotation bloqs deriving from `CirqGateAsBloqBase`. We can also follow the same pattern for 3 qubit gates instead of q1, q2, q3 etc.

Reply:
I'll make the change in the other pr's to be `ctrl` and `target`. Conceivably, that could be the case here as well; but CZ is symmetric.